x-pack/plugins/stack_connectors/server/connector_types/resilient/resilient.ts

```diff
@@ -0,0 +1,363 @@
+  ExecutorSubActionGetIncidentParamsSchema,
+  ExecutorSubActionGetIncidentTypesParamsSchema,
+  ExecutorSubActionGetSeverityParamsSchema,
+  ExecutorSubActionHandshakeParamsSchema,
+  GetCommonFieldsResponseSchema,
+  GetIncidentTypesResponseSchema,
+  GetSeverityResponseSchema,
+} from './schema';
+import { formatUpdateRequest } from './utils';
+
+const VIEW_INCIDENT_URL = `#incidents`;
+
+export class ResilientConnector extends CaseConnector<ResilientConfig, ResilientSecrets> {
+  private urls: {
+    incidentTypes: string;
+    incident: string;
+    comment: string;
+    severity: string;
+  };
+
+  constructor(params: ServiceParams<ResilientConfig, ResilientSecrets>) {
+    super(params);
+
+    this.urls = {
+      incidentTypes: `${this.getIncidentFieldsUrl()}/incident_type_ids`,
+      incident: `${this.getOrgUrl()}/incidents`,
+      comment: `${this.getOrgUrl()}/incidents/{inc_id}/comments`,
+      severity: `${this.getIncidentFieldsUrl()}/severity_code`,
+    };
+
+    this.registerSubActions();
+  }
+
+  protected getResponseErrorMessage(error: AxiosError) {
+    if (!error.response?.status) {
+      return i18n.UNKNOWN_API_ERROR;
+    }
+    if (error.response.status === 401) {
+      return i18n.UNAUTHORIZED_API_ERROR;
+    }
+    return `API Error: ${error.response?.statusText}`;
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT_TYPES,
+      method: 'getIncidentTypes',
+      schema: ExecutorSubActionGetIncidentTypesParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.SEVERITY,
+      method: 'getSeverity',
+      schema: ExecutorSubActionGetSeverityParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.FIELDS,
+      method: 'getFields',
+      schema: ExecutorSubActionCommonFieldsParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT,
+      method: 'getIncident',
+      schema: ExecutorSubActionGetIncidentParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.HANDSHAKE,
+      method: 'handshake',
+      schema: ExecutorSubActionHandshakeParamsSchema,
+    });
+  }
+
+  private getAuthHeaders() {
+    const token = Buffer.from(
+      this.secrets.apiKeyId + ':' + this.secrets.apiKeySecret,
+      'utf8'
+    ).toString('base64');
+
+    return { Authorization: `Basic ${token}` };
+  }
+
+  private getOrgUrl() {
+    const { apiUrl: url, orgId } = this.config;
+
+    return `${url}/rest/orgs/${orgId}`;
+  }
+
+  private getIncidentFieldsUrl = () => `${this.getOrgUrl()}/types/incident/fields`;
+
+  private getIncidentViewURL(key: string) {
+    const url = this.config.apiUrl;
+    const urlWithoutTrailingSlash = url.endsWith('/') ? url.slice(0, -1) : url;
+
+    return `${urlWithoutTrailingSlash}/${VIEW_INCIDENT_URL}/${key}`;
+  }
+
+  public async createIncident({
+    incident,
+  }: Record<string, unknown>): Promise<ExternalServiceIncidentResponse> {
```

Comment:
This is an implementation of abstract method of CaseConnector: 
`public abstract createIncident(
    incident: Record<string, unknown>
  ): Promise<ExternalServiceIncidentResponse>;`

So it throws error to use `{ incident: Incident }`.

Reply:
😢 It seems that the `CaseConnector` needs a lot of improvements.

Reply:
yeah but at the same time we need to keep it generic because other connectors will also use it in future. And not all those connectors will have same params.